arangod/Agency/MoveShard.cpp

```diff
@@ -279,28 +279,32 @@ bool MoveShard::start(bool&) {
   std::string planPath =
       planColPrefix + _database + "/" + _collection + "/shards/" + _shard;
   Slice planned = _snapshot.hasAsSlice(planPath).first;
-  TRI_ASSERT(planned.isArray());
 
   int found = -1;
   int foundTo = -1;
   int count = 0;
   _toServerIsFollower = false;
-  for (VPackSlice srv : VPackArrayIterator(planned)) {
-    TRI_ASSERT(srv.isString());
-    if (srv.copyString() == _to) {
-      foundTo = count;
-      if (!_isLeader) {
-        moveShardFinish(false,  false,
-               "toServer must not be planned for a following shard");
-        return false;
-      } else {
-        _toServerIsFollower = true;
+  if (planned.isArray()) {
+    for (VPackSlice srv : VPackArrayIterator(planned)) {
+      TRI_ASSERT(srv.isString());
+      if (srv.copyString() == _to) {
+        foundTo = count;
+        if (!_isLeader) {
+          moveShardFinish(false, false, "toServer must not be planned for a following shard");
+          return false;
+        } else {
+          _toServerIsFollower = true;
+        }
       }
+      if (srv.copyString() == _from) {
+        found = count;
+      }
+      ++count;
     }
-    if (srv.copyString() == _from) {
-      found = count;
-    }
-    ++count;
+  } else {
+    // We really should never be ever here
```

Comment:
I think we should adjust this code comment, as it is misleading. If we should *never* be here, why is this code triggered in some edge case? We should probably rephrase this to sth like "under normal circumstances we should not get here, but in the very unlikely case of ____ this code gets triggered"

Reply:
done